### DISCUSSION TITLE
Feature request: List all currently opened windows

### DISCUSSION THREAD

**Discussion by JCombee** [NativePHP/laravel#395] [Ideas]
For a app that I am working on I wan't to know which windows are already open.

## The Problem

My app I am working on is a monitor where I can monitor information for multiple projects.
Every project will get his own window.
In every window I also want a list of all projects.
If I already have this window open I want to visually already give it a different background color in that list and when you click it instead of opening a new window I want it to focus the window associated with the project.

## My idea for a solution

What I would think is a decent solution for this is having a extra function on the `Window` Facade that lists all the currently open windows.

Something like `Window::all();`. I do not know if this exact API is the right choice, but lets start the discussion.

**Comment by JCombee** [comment]
Looking into how the electron plugin is build I think just doing something similar as the window/current route works looks to me to be more consistent with the application.

I'll take a try in making a PR for this.

**Reply by JCombee** [reply]
- https://github.com/NativePHP/electron/pull/120
- https://github.com/NativePHP/laravel/pull/396

I was thinking that this would already be sufficient.

**Reply by simonhamp** [reply]
Would you be up for adding details to [the docs](/nativephp/nativephp.com) too?

**Comment by simonhamp** [comment]
Yep, this looks great 👍🏼
